scss/_patterns_logo-section.scss

```diff
@@ -0,0 +1,133 @@
+      @if $column-count == 1 {
+        // if the available number of columns of the parent container is less than the desired number of columns the logo should span, we clamp that desired number ($logo-column-span) to what's available:
+        $logo-column-span: min($logo-column-span, $column-count);
+      }
+
+      // The logos are sized so that, depending on breakpoint, they span either one, or two columns of the grid.
+      // When spanning more than one column, we need to account for the gutters between the columns.
+
+      // calc() doesn't allow dividing a mixture of different units, so the calculation
+      // (100% - $gutters-total-width) / $column-count needs to be broken down into two separate divisions:
+      // 100% / $column-count - $gutters-total-width / $column-count
+      // Thus the calculation is split into 2 parts, which are then used in the final calc() function:
+
+      // calculate the width of a single column by dividing the entire width by the number of columns it represents:
+      $percentage-width-of-one-col: 100% / $column-count;
+
+      // multiply the width of one column by the number of columns the logo needs to span
+      $percentage-contribution: $logo-column-span * $percentage-width-of-one-col;
+
+      // calculate the number of gutters the logo spans
+      $gutter-width-inside-logo: $gutter * ($logo-column-span - 1);
+
+      // add the width of gutters spanned by the logo, subtract the $gutters-total-width / $column-count;
+      // the signs here are reversed because the variable is subtracted. This avoids having + and - in the final css, e.g.: "+ -.75rem"
+      $rem-contribution: $logo-column-span * ($gutters-total-width / $column-count) - $gutter-width-inside-logo;
+
+      .#{$grid-column-prefix}#{$column-count} & {
+        @include vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution);
+      }
+    }
+  }
+}
+
+@mixin vf-p-apply-logo-section-item-width($percentage-contribution, $rem-contribution) {
+  @if $rem-contribution == 0 {
+    width: $percentage-contribution;
+  } @else {
+    width: calc(#{$percentage-contribution} - #{$rem-contribution});
+  }
+}
+
+@mixin vf-p-logo-section {
+  // The logo section pattern needs to align logos to the grid when they are an even number, and center them (while maintaining the same logo width) when the number of logos on a line is odd.
+  // This means we can't use css grid for this, instead we need to replicate its behavior for even numbers of logos, and center the odd ones.
+  .p-logo-section__title {
+    @extend %x-small-text;
+
+    font-weight: $font-weight-bold;
+    text-transform: uppercase;
+  }
+
+  .p-logo-section__items {
+    display: flex;
+    flex-wrap: wrap;
+    gap: 0 map-get($grid-gutter-widths, small);
+    justify-content: center;
+    width: 100%;
+
+    @media (min-width: $threshold-4-6-col) {
+      gap: 0 map-get($grid-gutter-widths, medium);
```

Comment:
`gap` on flexbox is only supported in latest Chrome/FF. This will not work for example in Safari:

<img width="742" alt="Screenshot 2021-04-08 at 12 33 37" src="https://user-images.githubusercontent.com/83575/114012534-bb379400-9866-11eb-85a3-8d8a761df6c3.png">

We may be fine with that (as it doesn't look broken), but we may need to use margins if we want it cross-browser.

Reply:
fixed